packages/documentation/src/stories/foundation/icons/icons.components.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import report from '@swisspost/design-system-icons/public/report.json';
+import './icons.styles.scss';
+
+const ICONS = report.icons.map(icon => Object.assign({}, icon, {
+  searchParameters: [icon.file.basename, ...icon.meta.keywords].join(' || ').toLowerCase()
+}));
+
+export class Search extends React.Component {
+  state = {
+    freetext: '',
+    icons: ICONS
+  };
+
+  freetextRef: React.RefObject<HTMLInputElement> = React.createRef();
+
+  searchFreetext (e: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({
+      freetext: e.target.value,
+      icons: ICONS
+        .filter(icon => icon.searchParameters.includes(e.target.value.toLowerCase()))
```

Comment:
The usage of localeCompare is not possible, because we want to find substrings.
Instead, I map normalized and lowercased keywords to the icon and search with a normalized and lowercased query.